exception_al/workshops/views.py

```diff
@@ -1,20 +1,71 @@
-from rest_framework.permissions import IsAuthenticated
+
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from .models import Workshop
 from .serializers import WorkshopSerializer
 from django.http import Http404
-from rest_framework import status
+from rest_framework import status, viewsets
+
 
 # Create your views here.
 # This is the view for creating a new workshop
 
-class WorkshopCreateView(APIView):
-    permission_classes = [IsAuthenticated]
+    
+    
+
+    
+    
+# This is the view for getting a list of all workshops
+    
+class WorkshopListView(APIView):
 
-    def post(self, request, format=None):
+    def get(self, request):
+        workshops = Workshop.objects.all()
+        serializer = WorkshopSerializer(workshops, many=True)
+        return Response(serializer.data)
+    
+    def post(self, request):
         serializer = WorkshopSerializer(data=request.data)
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    
+# This is the view for getting a single workshop by id    
+
+class WorkshopDetailView(APIView):
+
```

Comment:
Please add 
   permission_classes = [IsOwnerOrReadOnly]

We'll need to create a custom permission class in a new permissions.py file like this
```
from rest_framework import permissions

class IsOwnerOrReadOnly(permissions.BasePermission):
  def has_object_permission(self, request, view, obj):
    if request.method in permissions.SAFE_METHODS:
      return True
    return obj.owner == request.user
```

Reply:
Done! created permissions class accordingly.